chia/cmds/init_funcs.py

```diff
@@ -74,15 +79,31 @@ def check_keys(new_root: Path, keychain: Optional[Keychain] = None) -> None:
     all_targets = []
     stop_searching_for_farmer = "xch_target_address" not in config["farmer"]
     stop_searching_for_pool = "xch_target_address" not in config["pool"]
-    number_of_ph_to_search = 500
+    number_of_ph_to_search = 100
     selected = config["selected_network"]
     prefix = config["network_overrides"]["config"][selected]["address_prefix"]
+
+    intermediates = {}
+    for sk, _ in all_sks:
+        intermediates[bytes(sk)] = {
+            "observer": master_sk_to_wallet_sk_unhardened_intermediate(sk),
+            "non-observer": master_sk_to_wallet_sk_intermediate(sk),
+        }
+
     for i in range(number_of_ph_to_search):
         if stop_searching_for_farmer and stop_searching_for_pool and i > 0:
             break
         for sk, _ in all_sks:
+            intermediate_n = intermediates[bytes(sk)]["non-observer"]
+            intermediate_o = intermediates[bytes(sk)]["observer"]
+
+            all_targets.append(
+                encode_puzzle_hash(
+                    create_puzzlehash_for_pk(_derive_path_unhardened(intermediate_o, [i]).get_g1()), prefix
+                )
+            )
             all_targets.append(
-                encode_puzzle_hash(create_puzzlehash_for_pk(master_sk_to_wallet_sk(sk, uint32(i)).get_g1()), prefix)
+                encode_puzzle_hash(create_puzzlehash_for_pk(_derive_path(intermediate_n, [i]).get_g1()), prefix)
             )
             if all_targets[-1] == config["farmer"].get("xch_target_address"):
```

Comment:
Looks like the farmer and pool rewards addresses are only checked against the non-observer ph. Is that correct?

Reply:
Oh yeah we have to check both